### PR TITLE
[Docs] Update performance sections

### PR DESCRIPTION
**Pull request by faxu**
### Description
Staged: https://faxu.github.io/onnxruntime/docs/performance/

Main changes:
- Restructure performance section to break into sub-categories
- Move CUDA specific perf tuning tips to [CUDA EP page](https://faxu.github.io/onnxruntime/docs/execution-providers/CUDA-ExecutionProvider.html#performance-tuning)
- Update [Transformer optimizer page](https://faxu.github.io/onnxruntime/docs/performance/transformers-optimization.html) to remove version-specific content... will be supported along with https://github.com/microsoft/onnxruntime/pull/14964
- Fix links to point to new pages

### CODE REVIEW

**Review by cassiebreviu — APPROVED** [review]

File: docs/execution-providers/CUDA-ExecutionProvider.md [hunk 1]
```diff
@@ -85,23 +85,237 @@ Whether to do copies in the default stream or use separate streams. The recommen
...
 Default value: 0
 
+## Performance Tuning
+The [I/O Binding feature](../performance/tune-performance/iobinding.md) should be utilized to avoid overhead resulting from copies on inputs and outputs. 
+
+### Convolution-heavy models
+
+ORT leverages CuDNN for convolution operations and the first step in this process is to determine which "optimal" convolution algorithm to use while performing the convolution operation for the given input configuration (input shape, filter shape, etc.) in each `Conv` node. This sub-step involves querying CuDNN for a "workspace" memory size and have this allocated so that CuDNN can use this auxiliary memory while determining the "optimal" convolution algorithm to use. 
+
+By default, ORT clamps the workspace size to 32 MB which may lead to a sub-optimal convolution algorithm getting picked by CuDNN. To allow ORT to allocate the maximum possible workspace as determined by CuDNN, a provider option named `cudnn_conv_use_max_workspace` needs to get set (as shown below).
```
Comment (tianleiwu) [inline docs/execution-providers/CUDA-ExecutionProvider.md:112]: The default is max workspace in ORT 1.14.*. This section need update.

File: docs/performance/model-optimizations/ort-format-models.md [hunk 2]
```diff
@@ -117,34 +120,34 @@ python -m onnxruntime.tools.convert_onnx_models_to_ort --help
...
+    --custom_op_library CUSTOM_OP_LIBRARY
+                          Provide path to shared library containing custom operator kernels to register.
+    --save_optimized_onnx_model
+                          Save the optimized version of each ONNX model. This will have the same level of optimizations applied as the ORT format model.
+    --allow_conversion_failures
+                          Whether to proceed after encountering model conversion failures.
+    --nnapi_partitioning_stop_ops NNAPI_PARTITIONING_STOP_OPS
+                          Specify the list of NNAPI EP partitioning stop ops. In particular, specify the value of the "ep.nnapi.partitioning_stop_ops" session options config entry.
+    --target_platform {arm,amd64}
+                          Specify the target platform where the exported model will be used. This parameter can be used to choose between platform-specific options, such as QDQIsInt8Allowed(arm), NCHWc (amd64) and NHWC (arm/amd64) format, different optimizer level options, etc.
```
Comment (tianleiwu) [inline docs/performance/model-optimizations/ort-format-models.md:150]: what is the meaning of ` This parameter can be used to choose between platform-specific options, such as QDQIsInt8Allowed(arm), NCHWc (amd64) and NHWC (arm/amd64) format, different optimizer level options, etc.`

File: docs/execution-providers/CUDA-ExecutionProvider.md (hunk 1 above)
Comment (faxu) [reply docs/execution-providers/CUDA-ExecutionProvider.md:112]: @tianleiwu can you propose the revised text? 
We may also update content in a separate PR as there are content revisions needed across other pages as well. This is currently just copied over from existing documentation

File: docs/performance/model-optimizations/ort-format-models.md (hunk 2 above)
Comment (faxu) [reply docs/performance/model-optimizations/ort-format-models.md:150]: @skottmckay?

File: docs/execution-providers/CUDA-ExecutionProvider.md (hunk 1 above)
Comment (tianleiwu) [reply docs/execution-providers/CUDA-ExecutionProvider.md:112]: Example change:

The default value of `cudnn_conv_use_max_workspace` is 1 for versions 1.14 or later, and 0 for previous versions. When its value is 0, ORT clamps the workspace size to 32 MB which may lead to a sub-optimal convolution algorithm getting picked by CuDNN. To allow ORT to allocate the maximum possible workspace as determined by CuDNN, a provider option named `cudnn_conv_use_max_workspace` needs to get set (as shown below).

**Pull request merged**